app/src/main/java/org/sil/storyproducer/activities/BaseActivity.kt

```diff
@@ -60,16 +63,33 @@ open class BaseActivity : AppCompatActivity(), BaseActivityView {
     fun initWorkspace() {
         Workspace.initializeWorkspace(this)
         var showWelcome = true
-        if (Workspace.workdocfile.isDirectory) {
-            controller.updateStories()  // We already have a workspace folder
-            showWelcome = false
-        } else if (Workspace.createAppSpecificWorkspace()) {
-            controller.updateStories()  // Update after a new app-specific workspace was created
-            showWelcome = false
-        }
+        do {
+            if (Workspace.workdocfile.isDirectory) {
+                controller.updateStories()  // We already have a workspace folder
+                showWelcome = false
+                break
+            }
+            if (Workspace.workdocfile.uri.scheme == "file" &&
+                    Workspace.workdocfile.uri.path!!.isNotEmpty() &&
+                    Workspace.workdocfile.uri.path != "/") {
+                val wsStorageState = Environment.getExternalStorageState(File(Workspace.workdocfile.uri.path));
+                if (wsStorageState != Environment.MEDIA_MOUNTED) {
+                    // TODO: Replace 'Show media not mounted' toast message with a better welcome screen
+                    Toast.makeText(this, R.string.workspace_not_mounted, Toast.LENGTH_LONG).show()
```

Comment:
Replace with an error dialog. The expected storage device is not mounted. You can mount the storage device then restart Story Producer, or Start over with a new demo project. [Exit for now]  [Create new demo project]

Reply:
We can keep the current toast message for now as a more friendly approach has been noted in the "Automatic workspace folder" document and we can implement that when we have firmed up the ui and have more time.